main.nf

```diff
@@ -198,72 +249,328 @@ process get_software_versions {
 }
 
 
+/*
+ * STEP 1 - generate reversed decoy database
+ */
+process generate_decoy_database {
+    publishDir "${results_path}/"
+
+    input:
+     file fastafile
```

Comment:
This is not coming through a channel:

`file fastafile from ch_input_fasta`

Before that, you'd need to set up a channel that consumes the fasta file and produces [n] channels with the `.into (ch_1, ch_2)` operator to have the file appropriately linked in the process :-)